subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestOptionsIntegrationSpec.groovy

```diff
@@ -405,6 +407,179 @@ public class SomeIntegTestClass {
     }
     // endregion stand-alone test tasks
 
+    @Issue("https://github.com/gradle/gradle/issues/24331")
+    def "options can be set with framework before java plugin is applied"() {
+        given:
+        settingsFile << "rootProject.name = 'Sample'"
+        buildFile.text = """
+            tasks.withType(Test).configureEach {
+                useJUnitPlatform() {
+                    excludeTags = ["Slow"]
+                }
+                doFirst {
```

Comment:
🧐 Should `doFirst` be replaced with `doLast` to be more idiomatic?

Reply:
Got it.  Might be worth a note somewhere, to prevent a future drive-by "correction".